File: Array/T704_Binary Search/PYTHON/main.py
from typing import List
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1
        while left <= right:
            middle = left+((right-left)//2)
            if nums[middle] > target:
              right = middle - 1
            elif nums[middle]< target:
                left = middle+1
            else:
                return middle
        return -1
    

def test():
    nums = [-1,0,3,5,9,12]
    target = 9
    print("output:", Solution().search(nums, target))
    
test()
        

File: Array/T704_Binary Search/JS/main.js
var search = function(nums, target) {
    let mid,left=0, right = nums.length-1;
    while(left<=right){
        mid = left+((right - left) >> 1);
        if (nums[mid]>target) {
            right = mid-1;
        }else if (nums[mid]<target) {
            left = mid +1;
        }else{
            return mid;
        }

    }
    return -1;

};

let nums = [-1,0,3,5,9,12], target = 9;
console.log("output:", search(nums, target))